### PR TITLE
[GLUTEN-7087][CH] Support `WindowGroupLimitExec`

### PR DESCRIPTION
**Pull request by lgbo-ustc**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

Fixes: #7087 

`WindowGroupLimitExec` is introduced in spark-3.5. This PR makes it supported

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

unit tests

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: cpp-ch/local-engine/local_engine_jni.cpp [hunk 1]
```diff
@@ -166,6 +167,8 @@ JNIEXPORT jint JNI_OnLoad(JavaVM * vm, void * /*reserved*/)
     local_engine::BroadCastJoinBuilder::init(env);
     local_engine::CacheManager::initJNI(env);
 
+    DB::registerFormats();
```
Comment (baibaichen) [inline cpp-ch/local-engine/local_engine_jni.cpp:170]: Why does this PR need to call `DB:: tradeFormats()`?

File: cpp-ch/local-engine/Common/CHUtil.cpp [hunk 2]
```diff
@@ -315,6 +316,16 @@ DB::Block BlockUtil::concatenateBlocksMemoryEfficiently(std::vector<DB::Block> &
     return out;
 }
 
+String BlockUtil::dumpBlock(const DB::Block & block)
```
Comment (baibaichen) [inline cpp-ch/local-engine/Common/CHUtil.cpp:319]: How are about moving to `BlockTypeUtils.h/.cpp` ?

File: cpp-ch/local-engine/local_engine_jni.cpp (hunk 1 above)
Comment (lgbo-ustc) [reply cpp-ch/local-engine/local_engine_jni.cpp:170]: dump block needs formats.

Comment (lgbo-ustc) [reply cpp-ch/local-engine/local_engine_jni.cpp:170]: adding a simple method to dump block contents is very useful for debug

File: cpp-ch/local-engine/Common/CHUtil.cpp (hunk 2 above)
Comment (lgbo-ustc) [reply cpp-ch/local-engine/Common/CHUtil.cpp:319]: `BlockTypeUtils.h/.cpp` is just about types?

Comment (baibaichen) [reply cpp-ch/local-engine/Common/CHUtil.cpp:319]: Actually, it's block and type utils :)

File: cpp-ch/local-engine/local_engine_jni.cpp (hunk 1 above)
Comment (baibaichen) [reply cpp-ch/local-engine/local_engine_jni.cpp:170]: You mean calling `DB:: registerFormats()` is for later dumping?

File: cpp-ch/local-engine/Common/CHUtil.cpp (hunk 2 above)
Comment (baibaichen) [reply cpp-ch/local-engine/Common/CHUtil.cpp:319]: If dumping is just for debug aid, I think we should move it to `DebugUtils.h/cpp`, and put it to `debug` namespace. In product code, we should put it into` #ifdef NDEBUG ..  #endif`. e.g.

``` C++
#ifdef NDEBUG
 debug::dumpxxx();
#endif
```

**Review by baibaichen — APPROVED** [review]

**Pull request merged**